assets/js/base/components/cart-checkout/totals/shipping/utils.ts

```diff
@@ -31,3 +31,31 @@ export const getTotalShippingValue = ( values: {
 				parseInt( values.total_shipping_tax, 10 )
 		: parseInt( values.total_shipping, 10 );
 };
+
+/**
+ * Checks if no shipping methods are available or if all available shipping methods are local pickup
+ * only.
+ */
+export const areShippingMethodsMissing = (
+	hasRates: boolean,
+	prefersCollection: boolean | undefined,
+	shippingRates: CartResponseShippingRate[]
+) => {
+	if ( ! hasRates ) {
+		// No shipping methods available
+		return true;
+	}
+
+	// We check for the availability of shipping options if the shopper selected "Shipping"
+	if ( ! prefersCollection ) {
+		return shippingRates.some(
+			( shippingRatePackage ) =>
+				! shippingRatePackage.shipping_rates.some(
+					( shippingRate ) =>
+						shippingRate.method_id !== 'pickup_location'
```

Comment:
This will not work for third-party collectible rates, we need to check if the method ID is in `collectableRateIds` from the server.